src/storage/src/hummock/store/version.rs

```diff
@@ -213,44 +211,27 @@ impl HummockReadVersion {
                 StagingData::Sst(staging_sst) => {
-                    // TODO: enable this stricter check after each streaming table owns a read
-                    // version. assert!(self.staging.imm.len() >=
-                    // staging_sst.imm_ids.len()); assert!(staging_sst
-                    //     .imm_ids
-                    //     .is_sorted_by(|batch_id1, batch_id2| batch_id2.partial_cmp(batch_id1)));
-                    // assert!(
-                    //     check_subset_preserve_order(
-                    //         staging_sst.imm_ids.iter().cloned(),
-                    //         self.staging.imm.iter().map(|imm| imm.batch_id()),
-                    //     ),
-                    //     "the imm id of staging sstable info not preserve the imm order. staging
-                    // sst imm ids: {:?}, current imm ids: {:?}",
-                    //     staging_sst.imm_ids.iter().collect_vec(),
-                    //     self.staging.imm.iter().map(|imm| imm.batch_id()).collect_vec()
-                    // );
-                    // for clear_imm_id in staging_sst.imm_ids.iter().rev() {
-                    //     let item = self.staging.imm.back().unwrap();
-                    //     assert_eq!(*clear_imm_id, item.batch_id());
-                    //     self.staging.imm.pop_back();
-                    // }
-
-                    debug_assert!(
-                        check_subset_preserve_order(
-                            staging_sst.imm_ids.iter().cloned().sorted(),
-                            self.staging.imm.iter().map(|imm| imm.batch_id()).sorted()
-                        ),
-                        "the set of imm ids in the staging_sst {:?} is not a subset of current staging imms {:?}",
-                        staging_sst.imm_ids.iter().cloned().sorted().collect_vec(),
-                        self.staging.imm.iter().map(|imm| imm.batch_id()).sorted().collect_vec(),
-                    );
-
-                    let imm_id_set: HashSet<ImmId> =
-                        HashSet::from_iter(staging_sst.imm_ids.iter().cloned());
-                    self.staging
+                    let staging_imm_ids_from_sst: HashSet<u64> =
+                        staging_sst.imm_ids.iter().cloned().sorted().collect();
+                    let staging_imm_ids_from_imms: HashSet<u64> = self
+                        .staging
                         .imm
-                        .retain(|imm| !imm_id_set.contains(&imm.batch_id()));
-
-                    self.staging.sst.push_front(staging_sst);
+                        .iter()
+                        .map(|imm| imm.batch_id())
+                        .sorted()
+                        .collect();
+                    let intersection =
+                        staging_imm_ids_from_imms.intersection(&staging_imm_ids_from_sst);
+
+                    if intersection.count() > 0 {
+                        let imm_id_set: HashSet<ImmId> =
```

Comment:
Can we instead, keep `pop_back` if the imm id is in the imm id set, and after that assert all of the rest imm id is not in the imm id set?